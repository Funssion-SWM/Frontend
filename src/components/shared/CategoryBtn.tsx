type Props = {
  text: string;
  onClick: () => void;
  isSelected: boolean;
};

export default function CategoryBtn({ text, onClick, isSelected }: Props) {
  return (
    <button
      className={`border-2 texsomb20 py-2 px-4 rounded-3xl transition
          ${
            isSelected
              ? 'border-soma-blue-50 text-soma-blue-50 bg-soma-blue-10'
              : 'border-soma-grey-40 bg-white hover:border-soma-blue-50 hover:text-soma-blue-50 hover:bg-soma-blue-10'
          }`}
      onClick={() => onClick()}
    >
      {text}
    </button>
  );
}
