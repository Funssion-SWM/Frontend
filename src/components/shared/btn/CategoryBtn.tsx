import { ButtonSize } from '@/types';

type Props = {
  text: string;
  onClick: () => void;
  size?: ButtonSize;
  isSelected: boolean;
};

export default function CategoryBtn({
  text,
  onClick,
  size = 'medium',
  isSelected,
}: Props) {
  return (
    <button
      className={` py-1 px-3.5 transition ${
        {
          big: 'text-sm sm:text-base',
          medium: 'text-xs sm:text-sm',
          small: 'text-[10px] sm:text-xs',
        }[size]
      }
      ${
        isSelected
          ? 'border-soma-blue-50 text-soma-blue-50 font-semibold border-2 rounded-3xl'
          : 'border-soma-grey-40 bg-white hover:border-soma-blue-50 hover:text-soma-blue-50'
      }`}
      onClick={() => onClick()}
    >
      {text}
    </button>
  );
}
