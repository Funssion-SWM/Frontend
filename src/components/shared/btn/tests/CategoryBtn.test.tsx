import renderer from 'react-test-renderer';
import CategoryBtn from '../CategoryBtn';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('CategoryBtn', () => {
  it('renders correctly when isSelected is true', () => {
    const component = renderer.create(
      <CategoryBtn text="text" onClick={() => {}} isSelected={true} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly when isSelected is false', () => {
    const component = renderer.create(
      <CategoryBtn text="text" onClick={() => {}} isSelected={false} />
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly when size is small', () => {
    const component = renderer.create(
      <CategoryBtn
        text="text"
        size="small"
        onClick={() => {}}
        isSelected={false}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly when size is medium', () => {
    const component = renderer.create(
      <CategoryBtn
        text="text"
        size="medium"
        onClick={() => {}}
        isSelected={false}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders correctly when size is big', () => {
    const component = renderer.create(
      <CategoryBtn
        text="text"
        size="big"
        onClick={() => {}}
        isSelected={false}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('when button clicked', async () => {
    const mockCallback = jest.fn();

    render(
      <CategoryBtn text="text" onClick={mockCallback} isSelected={true} />
    );

    const btn = screen.getByRole('button');

    await userEvent.click(btn);
    expect(mockCallback.mock.calls.length).toEqual(1);
  });
});
