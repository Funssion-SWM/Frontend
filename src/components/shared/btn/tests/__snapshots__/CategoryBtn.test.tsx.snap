// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryBtn renders correctly when isSelected is false 1`] = `
<button
  className="border-2 texsomb20 py-2 px-4 rounded-3xl transition
          border-soma-grey-40 bg-white hover:border-soma-blue-50 hover:text-soma-blue-50 hover:bg-soma-blue-10"
  onClick={[Function]}
>
  text
</button>
`;

exports[`CategoryBtn renders correctly when isSelected is true 1`] = `
<button
  className="border-2 texsomb20 py-2 px-4 rounded-3xl transition
          border-soma-blue-50 text-soma-blue-50 bg-soma-blue-10"
  onClick={[Function]}
>
  text
</button>
`;
